tree/30_generic_methods/log_rudder.cf

```diff
@@ -280,16 +280,13 @@ bundle agent _rudder_common_report(technique_name, status, identifier, component
 #
 bundle agent _rudder_common_reports_generic(technique_name, class_prefix, identifier, component_name, component_key, message_prefix)
 {
-  classes:
-      "pass2" expression => "pass1";
-      "pass1" expression => "any";
+  methods:
 
-    pass2::
       # This case should NEVER happen. If it ever happens, it is a bug in CFEngine or ncf that lead to changing something in dry-run mode.
       # Hence, as we are facing a severe bug and we want to avoid changing more things, we define an abort class after displaying an error message .
-      "abort_agent_run" expression => "(dry_run|global_dry_run).${class_prefix}_repaired";
-
-  methods:
+      "abort" usebundle => _abort("repaired during dry-run", "Repaired previous component while in dry-run mode, this is a bug. Aborting immediately."),
```

Comment:
I'd put repaired_during_dryrun
Having underscore instead of spaces indicates an identifier that should not be modified.